js/reference.js

```diff
@@ -2797,6 +2850,35 @@
                 for (i = 0; i < compositeFKs.length; i++) {
                     this._referenceColumns.push(compositeFKs[i]);
                 }
+
+                // if array or true, add the remaining system columns
+                if (systemColumnsMode) {
+                    // array of column names to add
+                    var columnsToAdd = [];
+                    if (systemColumnsMode == true) {
+                        // add all, includes RID which will be skipped because of consideredColumns
+                        columnsToAdd = module._systemColumns;
+                    } else {
+                        // add the ones defined in array of config property
+                        // preserves order defined in `_systemColumns`
+                        columnsToAdd = module._systemColumns.filter(function (col) {
```

Comment:
This will ignore the ordering of `systemColumnsMode` and will order them based on `module._systemColumns`.

Reply:
Hongsuda and I discussed the ordering, and decided ordering won't be honored. If the property is set to a list (array), it will just add the columns defined in the array in a preset order (`RCB, RMB, RCT, RMT`) no matter what. `RID` will always be first, the rest in the above order at the end.

I should have updated the issue and mentioned this in the PR.